src/Symfony/Component/Uri/QueryString.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Uri;
+
+use Symfony\Component\Uri\Exception\InvalidQueryStringException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class QueryString implements \Stringable
+{
+    /**
+     * @var array<string, string|string[]>
+     */
+    private array $hashMap = [];
```

Comment:
Let's give this a better name. Like $parameters?

Reply:
Yes!